pkg/scheduler/framework/preemption/preemption.go

```diff
@@ -351,19 +357,19 @@ func (ev *Evaluator) prepareCandidate(ctx context.Context, c Candidate, pod *v1.
 			waitingPod.Reject(pluginName, "preempted")
 		} else {
 			if feature.DefaultFeatureGate.Enabled(features.PodDisruptionConditions) {
-				condition := &v1.PodCondition{
-					Type:    v1.AlphaNoCompatGuaranteeDisruptionTarget,
-					Status:  v1.ConditionTrue,
-					Reason:  "PreemptionByKubeScheduler",
-					Message: "Kube-scheduler: preempting",
-				}
-				newStatus := pod.Status.DeepCopy()
-				if apipod.UpdatePodCondition(newStatus, condition) {
-					if err := util.PatchPodStatus(ctx, cs, victim, newStatus); err != nil {
-						klog.ErrorS(err, "Preparing pod preemption", "pod", klog.KObj(victim), "preemptor", klog.KObj(pod))
-						errCh.SendErrorWithCancel(err, cancel)
-						return
-					}
+				victimPodApply := corev1apply.Pod(victim.Name, victim.Namespace).WithStatus(corev1apply.PodStatus())
+				victimPodApply.Status.WithConditions(corev1apply.PodCondition().
+					WithType(v1.AlphaNoCompatGuaranteeDisruptionTarget).
+					WithStatus(v1.ConditionTrue).
+					WithReason("PreemptionByKubeScheduler").
+					WithMessage("Kube-scheduler: preempting").
```

Comment:
Add "to accommodate a higher priority pod". Maybe we could even include the pod name?

Reply:
Added with the pod name / namespace. PTAL